pkg/skaffold/build/tag/env_template.go

```diff
@@ -50,7 +55,25 @@ func (t *envTemplateTagger) Labels() map[string]string {
 // GenerateFullyQualifiedImageName tags an image with the custom tag
 func (t *envTemplateTagger) GenerateFullyQualifiedImageName(workingDir string, opts Options) (string, error) {
 	customMap := CreateEnvVarMap(opts.ImageName, opts.Digest)
-	return util.ExecuteEnvTemplate(t.Template, customMap)
+	fqin, err := util.ExecuteEnvTemplate(t.Template, customMap)
+	if err != nil {
+		return "", err
+	}
+	return normalizeFQIN(fqin), nil
```

Comment:
I feel skaffold should not normalize the tag name. It should check if a tag is valid and then error out and maybe early on then at the final stage.
Not sure, everyone will agree that `skaffold`  changes all the invalid  image tags pattern under the hood to "-".
We do not want to enforce or suggest any tag naming conventions.

It would be great to fail early so users can change their env values to right one.
WDYT ?